base/ltspace.dtx

```diff
@@ -1217,7 +1218,23 @@
 \DeclareRobustCommand{\nobreakspace}{%
    \leavevmode\nobreak\ }
 \catcode `\~=13
-\def~{\nobreakspace{}}
+%</2ekernel>
+%<latexrelease>\IncludeInRelease{2023/11/01}%
+%<latexrelease>                 {[tilde]}{Protected tilde}%
+%<*2ekernel|latexrelease>
+\protected\edef~{%
+  \noexpand\ifincsname\noexpand\expandafter\string~%
+  \noexpand\else
+    \noexpand\expandafter\noexpand\nobreakspace
+  \noexpand\fi
+}
```

Comment:
OK I guess it's safe to drop the `{}` as `\protected` will mean will write as `~` to a file anyway. {perhaps a code level comment saying that would be useful) The expandafters are not really needed here, doing a nobreakspace inside an if doesn't hurt, but I guess it's good practice anyway.

Reply:
Yes, I originally had more `\expandafter` and kept the `{}`, but I think there were only there for the case where `~` expands to give `\nobreakspace .... [space] ...`, and we now don't have that issue. Happy to document.

I guess I'm used to having the tokens available to want to clean up conditionals all the time 'just in case'.

Reply:
Code comment added